packages/next-app/src/app/metadata/[network]/pool/[poolId]/(components)/TransactionModal.tsx

```diff
@@ -0,0 +1,227 @@
+"use client";
```

Comment:
I think this component is worth abstracting and putting into the `components` folder to be used by other apps. WDYT?

Reply:
It would be great ❤️